pkg/workceptor/controlsvc.go

```diff
@@ -182,13 +186,37 @@ func (t *workceptorCommandType) InitFromJSON(config map[string]interface{}) (con
 		if err != nil {
 			return nil, err
 		}
+		signature, err := strFromMap(config, "signature")
+		if err == nil {
+			c.params["signature"] = signature
+		}
 	}
 
 	return c, nil
 }
 
+func (c *workceptorCommand) processSignature(workType, signature string, connIsUnix bool) error {
+	shouldVerifySignature := c.w.ShouldVerifySignature(workType)
+	if !shouldVerifySignature && signature != "" {
+		return fmt.Errorf("work type did not expect a signature")
+	}
+	if shouldVerifySignature && !connIsUnix {
+		err := c.w.VerifySignature(signature)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
 // Worker function called by the control service to process a "work" command.
 func (c *workceptorCommand) ControlFunc(nc *netceptor.Netceptor, cfo controlsvc.ControlFuncOperations) (map[string]interface{}, error) {
+	addr := cfo.RemoteAddr()
+	connIsUnix := false
+	if addr.Network() == "unix" {
```

Comment:
@shanemcd we can do this to determine the origin of the connection to this control service.

if it's from a remote node on the mesh, Network will be "netceptor-foo". If it's from a tcp connection, it will be "tcp", and if from the local unix domain socket, "unix".

It should be safe to exempt signature verification in the case of unix socket. That is, we inherently trust any commands coming from the local domain socket.